presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```diff
@@ -88,6 +88,29 @@ public static JsonPath castCharToJsonPath(@LiteralParameter("x") Long charLength
         return new JsonPath(padSpaces(pattern, charLength.intValue()).toStringUtf8());
     }
 
+    @SqlNullable
+    @ScalarFunction("is_json_scalar")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean varcharIsJsonScalar(@SqlType("varchar(x)") Slice json)
+    {
+        return isJsonScalar(jsonParse(json));
+    }
+
+    @SqlNullable
+    @ScalarFunction
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean isJsonScalar(@SqlType(StandardTypes.JSON) Slice json)
+    {
+        try (JsonParser parser = createJsonParser(JSON_FACTORY, json)) {
+            JsonToken nextToken = parser.nextToken();
+            return nextToken != START_ARRAY && nextToken != START_OBJECT;
+        }
+        catch (IOException e) {
+            return null;
```

Comment:
Instead of returning null, we should thrown an `INVALID_CAST_ARGUMENT` instead, see functions in `JsonOperators` for example.

Reply:
Update: I see other functions in this class returns null on invalid json string. Let's discuss this next week.